### PR TITLE
Moved test out of run_in_graph_and_eager_mode in maxout.

### PR DESCRIPTION
**Pull request by gabrieldemarmiesse**
See #1328

### CODE REVIEW

File: tensorflow_addons/layers/maxout_test.py [hunk 1]
```diff
@@ -18,49 +18,45 @@
...
-            input_shape=(None, None, None, None),
-            input_data=inputs,
-        )
-
-    def test_invalid_shape(self):
-        with self.assertRaisesRegexp(ValueError, r"number of features"):
-            test_utils.layer_test(
-                Maxout, kwargs={"num_units": 3}, input_shape=(5, 4, 2, 7)
-            )
+pytestmark = pytest.mark.usefixtures("maybe_run_functions_eagerly")
```
Comment (AakashKumarNain) [inline tensorflow_addons/layers/maxout_test.py]: @gabrieldemarmiesse this isn't used anywhere else in this code. Is this required?

Comment (gabrieldemarmiesse) [reply tensorflow_addons/layers/maxout_test.py]: Yes, when you use a `pytestmark` variable in a test file run with pytest, the `pytest.mark.*` markers are applied  to all the test functions in the file. I agree that it's counter-intuitive. I like explicit stuff so I would have like it better if pytest had a function for that. Like `apply_to_all_tests_in_module(...)` or something like that. See http://doc.pytest.org/en/latest/example/markers.html#marking-whole-classes-or-modules

Comment (AakashKumarNain) [reply tensorflow_addons/layers/maxout_test.py]: Cool. Yeah that is not `pythonic` at all. Any other changes? If not, I will merge it

Comment (gabrieldemarmiesse) [reply tensorflow_addons/layers/maxout_test.py]: No other changes. :)

Comment (AakashKumarNain) [reply tensorflow_addons/layers/maxout_test.py]: Ok. Thanks

**Comment by gabrieldemarmiesse** [comment]
Thanks for the review!

**Comment by failure-to-thrive** [comment]
Too late, but looks like
```
from tensorflow_addons.utils import test_utils
```
is not required either...

**Comment by gabrieldemarmiesse** [comment]
@failure-to-thrive it's used in the last test.

Generally flake8 catches unused imports in the CI :)

**Pull request merged**